src/super_gradients/training/utils/quantization_utils.py

```diff
@@ -0,0 +1,384 @@
+    """
+    Activates quant modules wrapping, so that further modules use Q/DQ layers.
+    """
+    if _imported_pytorch_quantization_failure is not None:
+        raise _imported_pytorch_quantization_failure
+    else:
+        quant_modules.initialize()
+
+
+class QATCallback(PhaseCallback):
+    """
+    A callback for transitioning training into QAT.
+
+    Rebuilds the model with QAT layers then either:
+        1. loads the best checkpoint then performs calibration.
+        2. loads an external calibrated model (makes sense when start_epoch=0).
+
+    Additionally, resets SgModel's best_metric and sets ckpt_best_name to 'qat_ckpt_best.pth' so best QAT checkpoints
+     will be saved separately.
+
+    If performing calibration- the calibrated model is evaluated, and the metric_to_watch is logged under
+     calibrated_model_{metric_to_watch}. The calibrated checkpoint is saved under ckpt_calibrated_{calibration_method}.pth
+
+
+    Attributes:
+        start_epoch: int, first epoch to start QAT.
+
+        quant_modules_calib_method: str, One of [percentile, mse, entropy, max]. Statistics method for amax
+         computation of the quantized modules (default=percentile).
+
+        per_channel_quant_modules: bool, whether quant modules should be per channel (default=False).
+
+        calibrate: bool, whether to perfrom calibration (default=False).
+
+        calibrated_model_path: str, path to a calibrated checkpoint (default=None).
+
+        calib_data_loader: torch.utils.data.DataLoader, data loader of the calibration dataset. When None,
+         context.train_loader will be used (default=None).
+
+        num_calib_batches: int, number of batches to collect the statistics from.
+
+        percentile: float, percentile value to use when SgModel,quant_modules_calib_method='percentile'.
+         Discarded when other methods are used (Default=99.99).
+
+
+
+    """
+
+    def __init__(self, start_epoch: int, quant_modules_calib_method: str = "percentile",
+                 per_channel_quant_modules: bool = False, calibrate: bool = True, calibrated_model_path: str = None,
+                 calib_data_loader: DataLoader = None, num_calib_batches: int = 2, percentile: float = 99.99):
+        super(QATCallback, self).__init__(Phase.TRAIN_EPOCH_START)
+        self._validate_args(start_epoch, quant_modules_calib_method, calibrate, calibrated_model_path)
+        self.start_epoch = start_epoch
+        self.quant_modules_calib_method = quant_modules_calib_method
+        self.per_channel_quant_modules = per_channel_quant_modules
+        self.calibrate = calibrate
+        self.calibrated_model_path = calibrated_model_path
+        self.calib_data_loader = calib_data_loader
+        self.num_calib_batches = num_calib_batches
+        self.percentile = percentile
+
+    def _validate_args(self, start_epoch: int, quant_modules_calib_method: str, calibrate, calibrated_model_path):
+        if start_epoch < 0:
```

Comment:
test that the import worked as well. we don't want to run for 300 epochs and then crash on something we knew at the beginning

Reply:
Added